### PR TITLE
Add source fallback support for unsigned long mapped type

### PR DESCRIPTION
**Pull request by jdconrad**
This change adds source fallback support for unsigned long by adding a new `SourceValueFetcherSortedUnsignedLongIndexFieldData` similar to the other numeric types.

### CODE REVIEW

File: x-pack/plugin/mapper-unsigned-long/src/main/java/org/elasticsearch/xpack/unsignedlong/SourceValueFetcherSortedUnsignedLongIndexFieldData.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.elasticsearch.search.DocValueFormat.MASK_2_63;
+
+public class SourceValueFetcherSortedUnsignedLongIndexFieldData extends SourceValueFetcherIndexFieldData<SortedNumericDocValues> {
```
Comment (romseygeek) [inline x-pack/plugin/mapper-unsigned-long/src/main/java/org/elasticsearch/xpack/unsignedlong/SourceValueFetcherSortedUnsignedLongIndexFieldData.java:29]: Can we have some javadoc here?

File: x-pack/plugin/mapper-unsigned-long/src/main/java/org/elasticsearch/xpack/unsignedlong/SourceValueFetcherSortedUnsignedLongIndexFieldData.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        ) {
+            this.leafReaderContext = leafReaderContext;
+            this.valueFetcher = valueFetcher;
+            this.sourceLookup = sourceLookup;
+        }
+
+        @Override
+        public boolean advanceExact(int doc) throws IOException {
+            sourceLookup.setSegmentAndDocument(leafReaderContext, doc);
+            values = new ArrayList<>();
```
Comment (romseygeek) [inline x-pack/plugin/mapper-unsigned-long/src/main/java/org/elasticsearch/xpack/unsignedlong/SourceValueFetcherSortedUnsignedLongIndexFieldData.java:113]: Do we need to create a new list every time here or can we reuse an existing one?

**Review by romseygeek — COMMENTED** [review]
Thanks @jdconrad! I left a couple of comments.

File: x-pack/plugin/mapper-unsigned-long/src/main/java/org/elasticsearch/xpack/unsignedlong/SourceValueFetcherSortedUnsignedLongIndexFieldData.java (hunk 1 above)
Comment (jdconrad) [reply x-pack/plugin/mapper-unsigned-long/src/main/java/org/elasticsearch/xpack/unsignedlong/SourceValueFetcherSortedUnsignedLongIndexFieldData.java:35]: Added.

File: x-pack/plugin/mapper-unsigned-long/src/main/java/org/elasticsearch/xpack/unsignedlong/SourceValueFetcherSortedUnsignedLongIndexFieldData.java (hunk 2 above)
Comment (jdconrad) [reply x-pack/plugin/mapper-unsigned-long/src/main/java/org/elasticsearch/xpack/unsignedlong/SourceValueFetcherSortedUnsignedLongIndexFieldData.java]: Changed this to clear as suggested. Good catch.

**Comment by jdconrad** [comment]
@romseygeek Thank you the first review. I've updated the PR based on your feedback.

**Review by romseygeek — APPROVED** [review]
LGTM

**Pull request merged**